### PR TITLE
Support ranged/exact/oneof media constraints

### LINKED ISSUE #151: Support ranged media constraints
Pixel format is scored by complete match at now.
For example, I want to select one of `YUY2` and `UYVY` but not want to use `JPEG`.

https://github.com/pion/mediadevices/blob/949e850239b347a0750f005975caec7b8b77b5bb/pkg/prop/prop.go#L99-L112

### PR DESCRIPTION
**Pull request by at-wat**
### Reference issue
closes #151

### CODE REVIEW

File: pkg/prop/int.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+			return 0.0, true
+		}
+	}
+	return 1.0, false
+}
+
+type IntRanged struct {
+	Min   *int
+	Max   *int
+	Ideal *int
```
Comment (at-wat) [inline pkg/prop/int.go:40]: We are assuming zero value of constraint as not-set.

**Comment by lherman-cs** [comment]
@at-wat This is awesome work! It's kinda sad that there are many duplicated codes. But, I don't have an alternative either at this moment 😃.

I'm not really sure how to represent `Value`. I'm concerned with the safeness. Let's say that we want to get a value from an `IntConstraints`, the only safest way to get the value is to check first if the type is either `Int` or`IntExact`. But, this seems to be too implicit.

Maybe we can add a bool at the return value to tell the caller if the value can be represented? Similar to how go map getter works, `value, ok := someMap[key]`. If we take, `IntOneOf` as an example, it would be something like below:

```go
func (IntOneOf) Value() (int, bool) {
  return 0, false
}
```

What do you think?

**Comment by at-wat** [comment]
`Value() (int, bool)` looks better. Thanks!

File: pkg/codec/vaapi/vp8.go [hunk 2]
```diff
@@ -107,17 +107,32 @@ type encoderVP8 struct {
...
+	)
+	if width, ok = p.Width.Value(); !ok {
+		return nil, errors.New("media width is not constrained")
+	}
+	if height, ok = p.Height.Value(); !ok {
+		return nil, errors.New("media height is not constrained")
+	}
+	if frameRate, ok = p.FrameRate.Value(); !ok {
+		return nil, errors.New("media frame rate is not constrained")
+	}
```
Comment (at-wat) [inline pkg/codec/vaapi/vp8.go:123]: Maybe better to separate `prop.Media` (exact values) and `prop.MediaConstraint` (constraints)

**Pull request merged**